ui/src/components/auth/LoginContainer.tsx

```diff
@@ -0,0 +1,141 @@
+import React, { useState } from 'react';
+import { SubmitHandler, useForm } from 'react-hook-form';
+import { useDispatch } from 'react-redux';
+import { signin } from '../../reducers/userSlice';
+import axios from 'axios';
+import { Box, Button, Checkbox, Link, Stack, TextField, Typography } from '@mui/material';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { z } from 'zod';
+import { LoginFormFields } from './helpers/form.types';
+import { formSpacing, formWidth } from './helpers/constants';
+import { emailValidator, passwordValidator } from './helpers/zod.validators';
+
+// zod validation
+const loginValidationSchema = z.object({
+  email: emailValidator,
+  password: passwordValidator,
+});
+
+const LoginContainer: React.FC = () => {
+  const { register, handleSubmit, formState } = useForm<LoginFormFields>({
+    resolver: zodResolver(loginValidationSchema),
+  });
+
+  const { errors } = formState;
+
+  const [invalidLogin, setInvalidLogin] = useState<string>('');
+  const [rememberPassword, setRememberPassword] = useState<boolean>(false);
+
+  const dispatch = useDispatch();
+
+  const handleEmailLogin: SubmitHandler<LoginFormFields> = async (data) => {
+    try {
+      const payload = await axios.post('/api/auth/login', {
+        email: data.email,
+        password: data.password ?? '',
+        isRememberPassword: rememberPassword,
+      });
+      dispatch(signin(payload.data));
+      setInvalidLogin('');
+      window.location.reload();
```

Comment:
Why are we refreshing the page ya?

Reply:
i just followed the main web haha. I think it's because of how the PrivateRoute component in our main web was configured.

if (noUser) {
  if (noTokenExists) return <Component {...props} />;
  else {
    switch (props.location.pathname) {
      case '/login':
        if (user) {
          props.history.push('/profile');
          return <UserProfileContainer {...props} user={userObj} />;
        }
        break;
    }
    props.history.push('/');
    return <HomeContainer {...props} user={userObj} />;
  }
}

Assuming the user is not yet signed in, it wil fall under if (noTokenExists), hence the LoginContainer is returned. Once the user signs in, we reload the page and the next time PrivateRoute is called, it wil not go to the if (noTokenExists) branch anymore but instead go to the else { switch (props.location.pathname) } ... 

The successful sign in will dispatch a signin request, and the userObject will be stored in the redux store. Therefore the if (user) check is fulfilled and we go to props.history.push('/profile')

This is what I understand, do lmk if you have any thoughts haha

Reply:
oh hm makes sense, think its ok then just to make sure the login token is there